qrest/conf.py

```diff
@@ -369,6 +392,18 @@ def apply_default_headers(self, default):
         # re-validate to be sure current data is OK
         self.validate()
 
+    # ---------------------------------------------------------------------------------------------
+    def apply_default_timeout(self, default):
+        """Merge default set timeout on APIConfig level with local timeout settings
+        """
+
+        # Only apply the default if no timeout is set at the ResourceConfig level
+        if self.timeout is (0, 0):
+            self.timeout = default
```

Comment:
This piece of code made me realize that if you explicitly configure a `ResourceConfig` with timeout (0, 0), you will not be able to override it with the default that is set at the `APIConfig` level. Or am I missing / overlooking something?

One way out of this is to
1. set the default timeout of a `ResourceConfig` to `None` and
2. set the default timeout of a `APIConfig` to `(0, 0)`.

Lines 401 & 402 will become
``` python
if self.timeout is None:
    self.timeout = default
```
Probably this is not the only approach possible.

Reply:
Good point. The issue with the proposed solution is that you now have a default `timeout` value in a `ResourceConfig` that is not valid. I'm not familiar enough with `qrest` to be able to say that this will never cause problems. Someone that makes a `ResourceConfig` object directly might get in trouble here, but maybe it is the `qrest` philosophy to never do that (are we doing it in any of the tests maybe)?

By using `None` as a default for timeout as you propose, you open up the door for users to set `None` explicitly as well, which causes the same problem then.

This issue also translates to the headers. One can apply a default header on the `ApiConfig` level, and then have `headers=None` on the `ResourceConfig` level, in which case the default will stand.

A solution could be to track in the `ResourceConfig` init dunder whether the value was explicitly set when it was called. This can then be used to decide whether the `ApiConfig` defaults should be applied or not. Problem here is that there doesn't seem to exist an elegant way to track whether optional function parameters are explicitly set: https://stackoverflow.com/questions/14749328/how-to-check-whether-optional-function-parameter-is-set

Another approach could be to keep the behavior as is, and make it clear in the documentation.

Reply:
First a flash-forward to the comment by @JorisBenschop 
>  So the suggestion by Thomas to document instead has my preference here.

Although I'm not sure the rationale behind (t)his comment summarizes the underlying issue, I don't want to hold up the current pull request for this. Should it become a real-world issue, we'll tackle it then. So at least for now, let's just document it.

Now my 2 cents on your comment :)

> Someone that makes a ResourceConfig object directly

That's a valid point. I don't see anybody using a `ResourceConfig` directly but that might be my lack of imagination. And in case we use it like that in unit tests, we've implemented them ourselves so we should be able to fix them :)

> By using None as a default for timeout as you propose, you open up the door for users to set None explicitly as well, which causes the same problem then.

`None` is an unsupported value and should a user fill that in, it will be overwritten by the default value at the `APIConfig` level. That seems like acceptable behavior to me.

> This issue also translates to the headers.

Indeed, we can also fix the headers with this approach.

But again, let's just document the behavior and be done with it.

Reply:
> But again, let's just document the behavior and be done with it.

Done. How do you feel about the description in the documentation? Is it descriptive enough?